src/api/controllers/db-controller.ts

```diff
@@ -663,7 +663,7 @@ function parseDbTxTypeMetadata(dbTx: DbTx | DbMempoolTx): TransactionMetadata {
       let functionAbi: ClarityAbiFunction | undefined;
       const abi = dbTx.abi;
       if (abi) {
-        const contractAbi: ClarityAbi = JSON.parse(JSON.stringify(abi));
+        const contractAbi: ClarityAbi = typeof abi === 'string' ? JSON.parse(abi) : abi;
```

Comment:
@MHassanTariq do you think this could create problems? I added this because when retrieving the `abi` from the database, it comes as a JSON string already so there was no need to stringify it